### PR TITLE
Show Login on protected routes#2512

### LINKED ISSUE #2512: Show "You need to login.." for existing pages instead of 404
Pages that don't exist should have a 404 message, but pages that exist but can only be accessed when a user is logged in, should receive a message:

"Please login to access this page."

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2512

Changes: Show "You need to login.." for existing pages instead of 404

Demo Link: https://pr-2517-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 

Screenshot 2019-07-11 at 8 44 00 PM

### CODE REVIEW

File: src/components/shared/ProtectedRoute.js [hunk 1]
```diff
@@ -5,12 +5,16 @@ import { Route } from 'react-router-dom';
...
+  const { accessToken, isAdmin, component: Component, ...restProps } = props;
   return (
     <Route
       {...restProps}
       render={routeProps =>
-        accessToken ? <Component {...routeProps} /> : <NotFound />
+        accessToken ? (
+          <Component {...routeProps} />
+        ) : (
+          <NotFound renderLogin={!isAdmin} />
```
Comment (akshatnitd) [inline src/components/shared/ProtectedRoute.js]: Shouldn't render login be true always?

File: src/components/NotFound/NotFound.react.js [hunk 2]
```diff
@@ -16,14 +16,29 @@ const SusiLogo = styled.img.attrs({
...
+
+const NotFound = ({ accessToken, actions, renderLogin = false }) => {
   return (
     <div className="not-found-banner">
       <Link to={'/'}>
         <SusiLogo />
       </Link>
-      <h1>404</h1>
-      <h2>Page not found</h2>
+      <RenderText renderLogin />
```
Comment (akshatnitd) [inline src/components/NotFound/NotFound.react.js]: renderLogin={renderLogin}

File: src/components/shared/ProtectedRoute.js (hunk 1 above)
Comment (fragm3) [reply src/components/shared/ProtectedRoute.js]: Handling the case when user is not admin and tries to access admin panel

**Comment by fragm3** [comment]
@akshatnitd fixed

**Pull request merged**